### ISSUE TITLE
Should reject missing space after 'fun' when name in backticks

### ISSUE THREAD

**Issue by pkubowicz** [pinterest/ktlint#2879]
## Expected Behavior
There should always be a space after `fun` word:

https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods

Otherwise, ktlint should fail, like it fails with `Missing spacing after "if"`

IntelliJ formatter inserts a space after `fun`.

## Observed Behavior
ktlint accepts the code

## Steps to Reproduce
Put this to `src/test/kotlin`:

```kotlin
class MyClassTest {
    @Test
    fun`some test`() {
        println("hello world")
    }
}
```

editorconfig:

```
[{*.kt,*.kts}]
# ktlint_code_style=intellij_idea
```
(it does not matter which style you choose, in both cases no error happens)

and run ktlint

## Your Environment
* Version of ktlint used: 1.4.1

Called with `org.jlleitschuh.gradle.ktlint`.
